src/Consolonia.Themes/Templates/Controls/Helpers/ConsoleScrollContentPresenter.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Reflection;
+using Avalonia;
+using Avalonia.Controls.Presenters;
+using Avalonia.Input;
+using Avalonia.Media;
+using Avalonia.Utilities;
+
+namespace Consolonia.Themes.Templates.Controls.Helpers
+{
+    // This is workaround for Avalonia issue https://github.com/AvaloniaUI/AvaloniaEdit/issues/540
+    // Wheel mouse values are hard coded.
+    public class ConsoleScrollContentPresenter : ScrollContentPresenter
+    {
+        // Cache MethodInfo for private base method to avoid repeated reflection lookup.
+        private static readonly MethodInfo SnapOffsetMethod =
+            typeof(ScrollContentPresenter).GetMethod("SnapOffset",
+                BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.FlattenHierarchy);
+
+        protected override void OnPointerWheelChanged(PointerWheelEventArgs e)
+        {
+            // base.OnPointerWheelChanged(e);
+            if (Extent.Height > Viewport.Height || Extent.Width > Viewport.Width)
+            {
+                // var scrollable = Child as ILogicalScrollable;
```

Comment:
But this looks like breaking cases when child implements own scrolling.

Reply:
yeah, I backed off. I defer to child unless it's AvaloniaEdit (since we know it has a bug.)